File: packages/veil-sdk-dart/test/contact_bundle_test.dart
import "dart:typed_data";

import "package:test/test.dart";
import "package:veil_sdk/src/identity/contact_bundle.dart";
import "package:veil_sdk/src/identity/identity.dart";

void main() {
  test("contact bundle signs and verifies", () async {
    final identity = await generateIdentity();
    final bundle = ContactBundle(
      version: 1,
      pubkeyHex: identity.publicKeyHex,
      quicCertHex: "aa",
      endpoints: const ["quic://127.0.0.1:9000"],
      createdAt: 1,
    );
    final signed = await bundle.sign(identity);
    expect(signed.signatureHex, isNotNull);
    expect(await signed.verify(), isTrue);

    final encoded = signed.toCborBytes();
    final decoded = ContactBundle.fromCborBytes(encoded);
    expect(decoded.pubkeyHex, identity.publicKeyHex);
    expect(await decoded.verify(), isTrue);
  });
}

File: packages/veil-sdk-js/tests/contact_bundle.test.ts
import { describe, expect, it } from "vitest";
import {
  contactBundleFromQr,
  contactBundleToQr,
  decodeContactBundle,
  encodeContactBundle,
  generateIdentity,
  signContactBundle,
  verifyContactBundle,
} from "../src/identity";

describe("contact bundle", () => {
  it("signs and verifies", async () => {
    const identity = await generateIdentity();
    const bundle = {
      version: 1,
      pubkeyHex: Buffer.from(identity.publicKey).toString("hex"),
      quicCertHex: "aa",
      endpoints: ["quic://127.0.0.1:5000"],
      createdAt: 1,
    };
    const signed = await signContactBundle(bundle, identity);
    expect(await verifyContactBundle(signed)).toBe(true);

    const encoded = encodeContactBundle(signed);
    const decoded = decodeContactBundle(encoded);
    expect(decoded.pubkeyHex).toBe(bundle.pubkeyHex);
    expect(await verifyContactBundle(decoded)).toBe(true);

    const qr = contactBundleToQr(signed);
    const fromQr = contactBundleFromQr(qr);
    expect(fromQr.pubkeyHex).toBe(bundle.pubkeyHex);
  });
});
